File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/dto/search/SearchResult.js
/**
 * @author pkupczyk
 */
define([ "stjs" ], function(stjs) {
	var SearchResult = function() {
		this.objects = [];
		this.totalCount = 0;
	};
	stjs.extend(SearchResult, null, [], function(constructor, prototype) {
		prototype['@type'] = 'dto.search.SearchResult';
		constructor.serialVersionUID = 1;
		prototype.getObjects = function() {
			return this.objects;
		};
		prototype.getTotalCount = function() {
			return this.totalCount;
		}
	}, {});
	return SearchResult;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/common/search/SearchResult.java
/*
 * Copyright 2015 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search;

import java.io.Serializable;
import java.util.List;

import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.common.search.SearchResult")
public class SearchResult<OBJECT> implements Serializable
{

    private static final long serialVersionUID = 1L;

    private List<OBJECT> objects;

    private int totalCount;

    public SearchResult(List<OBJECT> objects, int totalCount)
    {
        this.objects = objects;
        this.totalCount = totalCount;
    }

    public List<OBJECT> getObjects()
    {
        return objects;
    }

    public int getTotalCount()
    {
        return totalCount;
    }

}
